repos/system_upgrade/el7toel8/actors/networkmanagerreadconfig/actor.py

```diff
@@ -0,0 +1,42 @@
+from leapp.actors import Actor
+from leapp.models import NetworkManagerConfig
+from leapp.tags import IPUWorkflowTag, FactsPhaseTag
+import subprocess
+import io
+
+try:
+    from configparser import ConfigParser
+except ImportError:
+    from ConfigParser import ConfigParser  # ver. < 3.0
+
+class NetworkManagerReadConfig(Actor):
+    name = 'network_manager_read_config'
+    description = 'This actor reads NetworkManager configuration.'
+    consumes = ()
+    produces = (NetworkManagerConfig,)
+    tags = (IPUWorkflowTag, FactsPhaseTag,)
+
+    def process(self):
+        try:
+            nm_config = NetworkManagerConfig()
+            # Use 'NM --print-config' to read the configurationo so
+            # that the main configuration file and other files in
+            # various directories get merged in the right way.
+            subp = subprocess.Popen(('/usr/sbin/NetworkManager', '--print-config'), stdout=subprocess.PIPE)
+            conf, err = subp.communicate()
+            parser = ConfigParser()
+            parser.readfp(io.StringIO(conf.decode('utf-8')))
+        except Exception as e:
```

Comment:
This is too wide, please try to restrict the exceptions handling first of all around the place where it could happen and secondly as restrictive as possible. e.g around the call there could be OSError, CalledProcessError. 

For the readfp part it should be ParsingError

Reply:
Ok.